### ISSUE TITLE
Using the pretrained models

### ISSUE THREAD

**Issue by shirAviv** [ox-vgg/vgg_face2#28]
Hi
I tried using your pretrained models with the VGG Face2 test dataset and i have two problems 
1. I don't understand how this could work for classification since the test trgates were never a part of the training and so no classification label was learned for it. is that correct?
2. for feature extraction, i followed the guidelines in your paper and site and generated template features by normaliztion. i then tried verifying by similarity of any single image of a person to all generated templates but the results were very low, around 150 out of 500 people were verified. do i need to do any preprocessing to the images before generating the features?
something else?

Thank you

**Comment by WeidiXie** [comment]
Hi, 

1. The model was trained with classification, and for verification, you simply need the feature encoder, so we remove the classifier in the model.

2. We have given the code for testing the model, if you want to customise the model for your specific application, please follow the pre-processing;
https://github.com/ox-vgg/vgg_face2/blob/9551679b70545e6bc4c9c88e98130ff1e5a16b34/standard_evaluation/ijb_evaluation.py#L1

Best,
Weidi

**Comment by shirAviv** [comment]
Hi Weidi
thanks for the quick reply. I would like to test classification on the test data of VGG-Face2, but looking at the output of the model i see that it returns both the features vector and a 8631 vector which i am assuming is the classification vector but only for the training subjects.
But this means that the model cannot classifiy except for subjects in the training data. is that correct?
Also, when i try to use it to classifiy images from the training data the classification values do not seem to match the actual class id of that subject as specified in the meta file. i guess i am a little confused in repsect to your pretrained models.
can you explain if i need to identify the faces first for the VGG Face2 dataset. (i don't have access to the IJB ones)

Thank you
Shir

Sent with [ProtonMail](https://protonmail.com) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by WeidiXie** [comment]
Hi, 

Face recognition is usually posed as an openset problem, meaning, during test time, you won't test on the images from identities you have trained on. Normally, you do verification, meaning compute similarity scores with respect to a gallery (where you know the identities of these images).

If you don't have IJB dataset, I would suggest you to test the verification on VGGFace2, for the images from same identity, test the cosine similarity.

Best,
Weidi

**Comment by shiyuanyin** [comment]
> 
> 
> Hi,
> 
>     1. The model was trained with classification, and for verification, you simply need the feature encoder, so we remove the classifier in the model.
> 
>     2. We have given the code for testing the model, if you want to customise the model for your specific application, please follow the pre-processing;
>        https://github.com/ox-vgg/vgg_face2/blob/9551679b70545e6bc4c9c88e98130ff1e5a16b34/standard_evaluation/ijb_evaluation.py#L1
> 
> 
> Best,
> Weidi

@WeidiXie 
I do not know the ijbb_face_tid_mid.txt and ijbb_template_pair_label.txt, how to generate

so I do not understand ,IJB Evaluation,  Could you explain it, please?

**Comment by WeidiXie** [comment]
@shiyuanyin 

Please refer to this:
https://www.nist.gov/programs-projects/face-challenges

**Comment by shiyuanyin** [comment]
------------------ 原始邮件 ------------------
发件人: "Weidi"<notifications@github.com>;
发送时间: 2019年6月25日(星期二) 下午4:36
收件人: "ox-vgg/vgg_face2"<vgg_face2@noreply.github.com>;
抄送: "史诗"<892781037@qq.com>;"Mention"<mention@noreply.github.com>;
主题: Re: [ox-vgg/vgg_face2] Using the pretrained models (#28)




@shiyuanyin
 
Please refer to this:
 https://www.nist.gov/programs-projects/face-challenges
 
—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.

Thank you for your reply.


Yes，before I'm asking for your advice，I have already read it,  but I still don't know how to generate  two txt file，these media id ,template ID and pair.txt , so I'd like to ask for your help。

**Comment by shiyuanyin** [comment]
> 
> 
> @shiyuanyin
> 
> Please refer to this:
> https://www.nist.gov/programs-projects/face-challenges
@WeidiXie 
Thank you for your reply.
Yes，before I'm asking for your advice，I have already read it,  but I still don't know how to generate  two txt file，these media id ,template ID and pair.txt , so I'd like to ask for your help。

**Comment by WeidiXie** [comment]
You don't have to use the exactly same txt files. 

These are just the way we used, you can follow the data structure from NIST, and test the model in your own way.

In our case, the image has :
Media ID:  which video or image the face come from.
Template ID: which template this image belongs to.

The goal is then to do verification for the template pairs.

**Comment by shiyuanyin** [comment]
> 
> 
> You don't have to use the exactly same txt files.
> 
> These are just the way we used, you can follow the data structure from NIST, and test the model in your own way.
> 
> In our case, the image has :
> Media ID: which video or image the face come from.
> Template ID: which template this image belongs to.
> 
> The goal is then to do verification for the template pairs.

@WeidiXie 
Thank you ,I understand now
